internal/scheduler/reports.go

```diff
@@ -31,19 +31,27 @@ type SchedulingContextRepository struct {
 	mostRecentSchedulingContextByExecutorP atomic.Pointer[SchedulingContextByExecutor]
 	// The most recent attempt where a non-zero amount of resources were scheduled.
 	mostRecentSuccessfulSchedulingContextByExecutorP atomic.Pointer[SchedulingContextByExecutor]
+	// The most recent attempt that preempted at least one job.
+	mostRecentPreemptingSchedulingContextByExecutorP atomic.Pointer[SchedulingContextByExecutor]
+
 	// Maps queue name to QueueSchedulingContextByExecutor.
 	// The most recent attempt.
 	mostRecentQueueSchedulingContextByExecutorByQueueP atomic.Pointer[map[string]QueueSchedulingContextByExecutor]
 	// The most recent attempt where a non-zero amount of resources were scheduled.
 	mostRecentSuccessfulQueueSchedulingContextByExecutorByQueueP atomic.Pointer[map[string]QueueSchedulingContextByExecutor]
+	// Map from queue name to most recent attempt that preempted at least one job belonging to this queue.
+	mostRecentPreemptingQueueSchedulingContextByExecutorByQueueP atomic.Pointer[map[string]QueueSchedulingContextByExecutor]
+
 	// Maps job id to JobSchedulingContextByExecutor.
 	// We limit the number of job contexts to store to control memory usage.
 	mostRecentJobSchedulingContextByExecutorByJobId *lru.Cache
+
```

Comment:
I can be talked out of making these whitespace changes, but I found it helpful to group the fields like this; same for the functions below.